tests/Domain/Integrations/Controllers/IntegrationControllerTest.php

```diff
@@ -5,39 +5,130 @@
 use App\Domain\Coupons\Coupon;
 use App\Domain\Coupons\Models\CouponModel;
+use App\Domain\Integrations\Environment;
 use App\Domain\Integrations\Integration;
 use App\Domain\Integrations\IntegrationPartnerStatus;
 use App\Domain\Integrations\IntegrationStatus;
 use App\Domain\Integrations\IntegrationType;
+use App\Domain\Integrations\IntegrationUrl;
+use App\Domain\Integrations\IntegrationUrlType;
 use App\Domain\Integrations\Models\IntegrationModel;
+use App\Domain\Integrations\Models\IntegrationUrlModel;
 use App\Domain\Organizations\Address;
 use App\Domain\Organizations\Models\OrganizationModel;
 use App\Domain\Organizations\Organization;
+use App\ProjectAanvraag\ProjectAanvraagUrl;
 use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Validation\UnauthorizedException;
 use Ramsey\Uuid\Uuid;
 use Tests\TestCase;
 
 final class IntegrationControllerTest extends TestCase
 {
     use RefreshDatabase;
 
+    public function test_it_can_store_an_integration(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
+
+        $subscriptionId = Uuid::uuid4();
+
+        $response = $this->post(
+            '/integrations',
+            [
+                'integrationType' => IntegrationType::SearchApi->value,
+                'subscriptionId' => $subscriptionId->toString(),
+                'integrationName' => 'Test Integration',
+                'description' => 'Test Integration description',
+                'firstNameFunctionalContact' => 'John',
+                'lastNameFunctionalContact' => 'Doe',
+                'emailFunctionalContact' => 'john.doe@test.com',
+                'firstNameTechnicalContact' => 'John',
+                'lastNameTechnicalContact' => 'Doe',
+                'emailTechnicalContact' => 'john.doe@test.com',
+                'agreement' => 'true',
+                'privacy' => 'some privacy',
+            ]
+        );
+
+        $response->assertRedirect('/nl/integraties');
+
+        $this->assertDatabaseHas('integrations', [
+            'type' => IntegrationType::SearchApi->value,
+            'subscription_id' => $subscriptionId->toString(),
+            'name' => 'Test Integration',
+            'description' => 'Test Integration description',
+            'status' => IntegrationStatus::Draft->value,
+            'partner_status' => IntegrationPartnerStatus::THIRD_PARTY->value,
+        ]);
+
+        $this->assertDatabaseHas('contacts', [
+            'type' => ContactType::Functional,
+            'first_name' => 'John',
+            'last_name' => 'Doe',
+        ]);
+
+        $this->assertDatabaseHas('contacts', [
+            'type' => ContactType::Technical,
+            'first_name' => 'John',
+            'last_name' => 'Doe',
+        ]);
+    }
+
+    public function test_it_can_destroy_an_integration(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
```

Comment:
The second parameter of `actingAs` is not the middleware group to use but the guard. So I think it can be removed here.

Reply:
Done